torch/_refs/nn/functional/__init__.py

```diff
@@ -542,6 +543,38 @@ def l1_loss(
     return _apply_loss_reduction(loss, reduction)
 
 
+@elementwise_type_promotion_wrapper(
+    type_promoting_args=("input", "target"),
+    type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.COMPLEX_TO_FLOAT,
+)
+def smooth_l1_loss(
+    input: TensorLikeType,
+    target: TensorLikeType,
+    size_average: Optional[bool] = None,
+    reduce: Optional[bool] = None,
+    reduction: str = "mean",
+    beta: float = 1.0,
+) -> TensorLikeType:
+    """
+    Reference implementation of torch.nn.functional.smooth_l1_loss
+    """
+    if size_average is not None or reduce is not None:
+        # TODO: Raise exception instead of converting value.  This is only for
+        # primTorch since it can drop support for deprecated arguments.
+        # msg = "size_average and reduce args are deprecated, please use reduction argument."
+        reduction = _get_string_reduction_arg(size_average=size_average, reduce=reduce)
+    _check_reduction_value(reduction)
+
+    if beta == 0.0:
```

Comment:
why do you need this conditional?

Reply:
Python nn functional API does the same: https://github.com/pytorch/pytorch/blob/76af22103b4b886ca5d124d1df2e6cf8f53065f3/torch/nn/functional.py#L3242-L3245

Reply:
I don't think we have enough logic as to optimise this example out in practice, as it would mean that we need to prove that `(input-target).abs()` is not negative. The conditional is alright for now.

Reply:
wdym? `.abs()` is non-negative. Functional API does this due to some numeric discrepancies in backward, this doesn't apply here.

Reply:
Ok, my point simply comes from a perf perspective, wheere we would be computing both branches of the where and just using one, but probably LLVM should be able to catch the ` < 0` after `.abs()` and optimise it out.

That being said, I still think that keeping this closer to core is better, as we could think of eventually registering this operation and simply differentiating through it to get its backward. This `beta==0` specialisation would make sure that this works in that case, as it does in master.